File: 30DaysLeetCodingChallenge/MaximumSubarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 0) {
            return Integer.MIN_VALUE;
        }
        if (nums.length == 1) {
            return nums[0];
        }
        int res = nums[0];
        int curr_max = nums[0];
        for (int i = 1; i < nums.length; i++) {
            curr_max = Math.max(curr_max + nums[i], nums[i]);
            res = Math.max(res, curr_max);
        }
        return res;
    }
}

File: 30DaysLeetCodingChallenge/MaximumSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if(nums.size() == 0){
            return INT_MIN;
        }
        if(nums.size() == 1){
            return nums[0];
        }
        int res = nums[0];
        int curr_max = nums[0];
        for(int i = 1; i<nums.size();i++){
            curr_max = max(nums[i], nums[i] + curr_max);
            res = max(res, curr_max);
        }
        return res;
    }
};